Makefile

```diff
@@ -15,137 +15,72 @@ BUILD=`git rev-parse HEAD`
-TAG=${IMAGE}:${VERSION}
+
+# Lighthouse Docker settings.
+LH_TAG=${LH_IMAGE}:${VERSION}
 
 # GO settings.
-EXECNAME=lh-server
-BINARY=bin/${EXECNAME}
 GOOS=linux
 GO=`which go`
 GOBUILD=CGO_ENABLED=0 GOOS=${GOOS} ${GO} build
 GOTEST=${GO} list -f '{{if len .TestGoFiles}}"go test -cover {{.ImportPath}}"{{end}}' ./... | xargs -L 1 sh -c
-PACKAGEPATH=./cmd/lh-server/...
+
+# Lighthouse GO settings.
+LH_BIN_PATH=bin/lighthouse-server
+LH_PKG_PATH=./cmd/lighthouse-server/...
 
 # Setup -ldflags for go build.
 # Allows setting some global vars before compilation.
 LDFLAGS=-ldflags "-X main.Version=${VERSION} -X main.Build=${BUILD}"
 
 # Show available make commands.
-default:
+usage:
 	@echo "Please supply one of:"
-	@echo "\tconfig\t\t- Set GCP configurations."
-	@echo "\tcreds\t\t- Get GKE credentials."
-	@echo "\tdeps\t\t- Install dependencies."
-	@echo "\tbuild.bin\t- Build the GO binary."
-	@echo "\tclean.bin\t- Clean the GO binary."
-	@echo "\tbuild.image\t- Build the Docker image."
-	@echo "\tbuild.run.image\t- Rebuild & run the Docker image with docker-compose up."
-	@echo "\trun.image\t- Run the Docker image with docker-compose up."
-	@echo "\tstop.image\t- Stop the Docker image with docker-compose down."
-	@echo "\tpush.image\t- Push the Docker image to GCR."
-	@echo "\tclean.image\t- Clean the Docker image from the host machine."
-	@echo "\tbuild.cluster\t- Build the GKE cluster."
-	@echo "\tdeploy.cluster\t- Deploy the GKE cluster."
-	@echo "\tclean.cluster\t- Clean the GKE cluster."
-	@echo "\ttest\t\t- Run the go test suite."
+	@echo "\tdeps:\n\t\t- Install dependencies."
+	@echo "\tconfig:\n\t\t- Set GCP configurations."
+	@echo "\tbuild.bins\n\t\t- Build all the GO binaries."
+	@echo "\tclean.bins\n\t\t- Clean all the GO binaries."
+	@echo "\tbuild.images\n\t\t- Build all the Docker images."
+	@echo "\tbuild.up\n\t\t- Rebuild & run the Docker images with docker-compose up."
+	@echo "\tup\n\t\t- Run the Docker images with docker-compose up."
+	@echo "\tdown\n\t\t- Stop the Docker images with docker-compose down."
+	@echo "\ttest:\n\t\t- Run the GO test suite."
+	@make lighthouse.usage
+
+# Include Makefiles.
+include docker/lighthouse-server/Makefile
+
+# Install dependencies.
+deps:
+	@echo "Installing dependencies ..."
+	@glide install
 
 # Set GCP configurations.
 config:
 	@gcloud config set project ${PROJECT}
 	@gcloud config set compute/zone ${ZONE}
 	@gcloud config set container/new_scopes_behavior true
 
-# Get GKE credentials.
-creds:
-	@gcloud container clusters get-credentials ${CLUSTER}
+# Build all the GO binaries.
+build.bins: lighthouse.build.bin
 
-# Install dependencies.
-deps:
-	@echo "Installing dependencies ..."
-	@glide install
+# Clean all the GO binaries.
+clean.bins: lighthouse.clean.bin
 
-# Build the GO binary.
-build.bin:
-	@echo "Building ${BINARY} ..."
-	@${GOBUILD} ${LDFLAGS} -o ${BINARY} ${PACKAGEPATH}
+# Build all the Docker images.
+build.images: deps clean.bins build.bins lighthouse.build.image
```

Comment:
This looks like it could get out of control. Its fine for now I think, but I wonder if there is a way we could improve on it.

Reply:
We can also do this to clean it up later, but agreed it could get out of hand if there are many images to build.
```
build.images: deps clean.bins build.bins
    @make lighthouse.build.image
    @...
```